File: 230118-dojo/java/app/src/main/java/Solution.java
class Solution {
    public long solution(int price, int money, int count) {
        long remain = money;

        for (int i = 1; i <= count; i += 1) {
            remain -= (long) price * i;
        }

        return remain < 0 ? Math.abs(remain) : 0;
    }
}

File: 230118-dojo/javascript/solution.js
export default function solution(price, money, count) {
  const totalPrice = new Array(count).fill(0)
    .map((_, index) => price * (index + 1))
    .reduce((pre, cur) => pre + cur, 0);

  return money - totalPrice >= 0 ? 0 : -(money - totalPrice);
}
